nilearn/plotting/surface/_backend.py

```diff
@@ -0,0 +1,434 @@
+            "'plotly'."
+        )
+
+
+class SurfaceBackend:
+    def plot_surf(
+        self,
+        surf_mesh=None,
+        surf_map=None,
+        bg_map=None,
+        hemi=None,
+        view=None,
+        cmap=None,
+        symmetric_cmap=None,
+        colorbar=True,
+        avg_method=None,
+        threshold=None,
+        alpha=None,
+        bg_on_data=False,
+        darkness=0.7,
+        vmin=None,
+        vmax=None,
+        cbar_vmin=None,
+        cbar_vmax=None,
+        cbar_tick_format="auto",
+        title=None,
+        title_font_size=None,
+        output_file=None,
+        axes=None,
+        figure=None,
+    ):
+        check_params(locals())
+        if view is None:
+            view = "dorsal" if hemi == "both" else "lateral"
+
+        surf_map, surf_mesh, bg_map = check_surface_plotting_inputs(
+            surf_map, surf_mesh, hemi, bg_map
+        )
+
+        check_extensions(surf_map, DATA_EXTENSIONS, FREESURFER_DATA_EXTENSIONS)
+
+        coords, faces = load_surf_mesh(surf_mesh)
+
+        fig = self._plot_surf(
+            coords,
+            faces,
+            surf_map=surf_map,
+            bg_map=bg_map,
+            hemi=hemi,
+            view=view,
+            cmap=cmap,
+            symmetric_cmap=symmetric_cmap,
+            colorbar=colorbar,
+            avg_method=avg_method,
+            threshold=threshold,
+            alpha=alpha,
+            bg_on_data=bg_on_data,
+            darkness=darkness,
+            vmin=vmin,
+            vmax=vmax,
+            cbar_vmin=cbar_vmin,
+            cbar_vmax=cbar_vmax,
+            cbar_tick_format=cbar_tick_format,
+            title=title,
+            title_font_size=title_font_size,
+            output_file=output_file,
+            axes=axes,
+            figure=figure,
+        )
+
+        return fig
+
+    def plot_surf_contours(
+        self,
+        surf_mesh=None,
+        roi_map=None,
+        hemi=None,
+        levels=None,
+        labels=None,
+        colors=None,
+        legend=False,
+        cmap="tab20",
+        title=None,
+        output_file=None,
+        axes=None,
+        figure=None,
+        **kwargs,
+    ):
+        # TODO hemi returns None from here, if I pass to plot_surf,
+        # returns error
+        hemi = sanitize_hemi_for_surface_image(hemi, roi_map, surf_mesh)
+        roi_map, surf_mesh, _ = check_surface_plotting_inputs(
+            roi_map, surf_mesh, hemi, map_var_name="roi_map"
+        )
+        check_extensions(roi_map, DATA_EXTENSIONS, FREESURFER_DATA_EXTENSIONS)
+
+        fig = self._plot_surf_contours(
+            surf_mesh=surf_mesh,
+            roi_map=roi_map,
+            levels=levels,
+            labels=labels,
+            colors=colors,
+            legend=legend,
+            cmap=cmap,
+            title=title,
+            output_file=output_file,
+            axes=axes,
+            figure=figure,
+            **kwargs,
+        )
```

Comment:
Shall I set `hemi='left'` if `hemi=None`? 

It is not done explicitly in the original code, it is done implicitly by not passing `hemi` value to `plot_surf` function and `plot_surf` uses its default which is `hemi=left`. But I think it is better to do it explicitly; ie. either set `hemi='left'` (or other valid value) or handle the error when `hemi=None`.

Reply:
yes let's do that but it'd be good to centralize those defaults in a single place otherwise we end up having to manual check they are consistent in an increasing number of places

Reply:
I can set all defaults at the top of `surf_plotting`. However, if the user enters `hemi=None`, then there is an error. Then does that mean that `hemi` should never have value `None`?